packages/snaps-controllers/src/services/webview/WebViewExecutionService.ts

```diff
@@ -1,32 +1,35 @@
-import type { ExecutionServiceArgs } from '../AbstractExecutionService';
-import { ProxyExecutionService } from '../proxy/ProxyExecutionService';
+import type { TerminateJobArgs } from '../AbstractExecutionService';
+import {
+  AbstractExecutionService,
+  type ExecutionServiceArgs,
+} from '../AbstractExecutionService';
 import type { WebViewInterface } from './WebViewMessageStream';
 import { WebViewMessageStream } from './WebViewMessageStream';
 
 export type WebViewExecutionServiceArgs = ExecutionServiceArgs & {
-  getWebView: () => Promise<WebViewInterface>;
+  createWebView: (jobId: string) => Promise<WebViewInterface>;
+  removeWebView: (jobId: string) => void;
 };
 
-export class WebViewExecutionService extends ProxyExecutionService {
-  #getWebView;
+export class WebViewExecutionService extends AbstractExecutionService<string> {
```

Comment:
Shouldn't this be `WebViewInterface`?

```suggestion
export class WebViewExecutionService extends AbstractExecutionService<WebViewInterface> {
```

Reply:
Good point, yeah. It doesn't matter though since we just use the ID to tear it down. Will change regardless